### PR TITLE
before:browser:launch api changes

### PR DESCRIPTION
**Pull request by Bkucera**
changes for https://github.com/cypress-io/cypress/pull/6293

- [x] fix example modifying firefox args 
- [x] add firefox prefs example

### CODE REVIEW

File: source/api/plugins/browser-launch-api.md [hunk 1]
```diff
@@ -10,23 +10,25 @@ The most common use case is adding your own web extension - {% url "example reci
...
-`args` may be an array or an object (based on the type of browser we're launching). Whatever you return from this event will become the new args for launching the browser.
+The returned `options` object will become the new launch options for the browser.
 
 Here are options for the currently supported browsers:
 
 * {% url 'Chrome, Chromium, Canary, or Microsoft Edge (Chromium based) browsers' "https://peter.sh/experiments/chromium-command-line-switches/" %}
 * {% url 'Firefox' "http://kb.mozillazine.org/About:config_entries" %}
 * {% url 'Electron' "https://github.com/electron/electron/blob/master/docs/api/browser-window.md#new-browserwindowoptions" %}
 
+Modify browser launch arguments:
```
Comment (flotwig) [inline source/api/plugins/browser-launch-api.md]: these should probably be headings of some sort, right?

File: source/api/plugins/browser-launch-api.md [hunk 2]
```diff
@@ -55,10 +57,37 @@ module.exports = (on, config) => {
...
 
-      // whatever you return here becomes the new args
-      return args
+      return options
+    }
+  })
+}
+```
+
+Add browser extensions:
```
Comment (flotwig) [inline source/api/plugins/browser-launch-api.md]: ditto

File: source/api/plugins/browser-launch-api.md [hunk 3]
```diff
@@ -55,10 +57,57 @@ module.exports = (on, config) => {
...
+  })
+}
+```
+
+Changing browser preferences:
+
+```js
+module.exports = (on, config) => {
+  on('browser:before:launch', (browser, options) => {
+    if (browser.family === 'chromium' && browser.name !== 'electron') {
```
Comment (brian-mann) [inline source/api/plugins/browser-launch-api.md]: worth calling out that this targets all `chrome`, `chromium`, `canary`, `edge`, etc browser versions but excludes `electron`

File: source/guides/references/migration-guide.md [hunk 4]
```diff
@@ -169,6 +169,28 @@ spy.resetHistory()
...
+
+Since we now support more advanced browser launch options, during `before:browser:launch` we no longer yield the second argument as an array of browser arguments and instead yield an `options` object with an `args` property.
+
+You can see more examples of the new `options` in use in the {% url "Browser Launch API doc" browser-launch-api %}.
+
+{% badge danger Before %} The second argument is no longer an array.
+
+```js
+on('browser:before:launch', (browser, args) => {
+  args.push('--another-arg')
```
Comment (brian-mann) [inline source/guides/references/migration-guide.md]:
```suggestion
  // will print a deprecation warning telling you
  // to change your code to the new signature
  args.push('--another-arg')
```

File: source/api/plugins/browser-launch-api.md (hunk 3 above)
Comment (jennifer-shehane) [reply source/api/plugins/browser-launch-api.md]: Well, this check is used in several places so I'm not sure we want to list out all browsers here necessarily.

**Comment by jennifer-shehane** [comment]
I'm merging this into 4.0. Any changes that are still required for this PR should be made to https://github.com/cypress-io/cypress-documentation/pull/1710 directly.

**Pull request merged**